rust/noosphere/src/storage.rs

```diff
@@ -42,22 +42,22 @@ impl From<StorageLayout> for PathBuf {
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-use noosphere_storage::{NativeStorage, NativeStorageInit};
+use noosphere_storage::{SledStorage, SledStorageInit};
 
 #[cfg(not(target_arch = "wasm32"))]
 impl StorageLayout {
-    pub async fn to_storage(&self) -> Result<NativeStorage> {
-        NativeStorage::new(NativeStorageInit::Path(PathBuf::from(self)))
+    pub async fn to_storage(&self) -> Result<SledStorage> {
+        SledStorage::new(SledStorageInit::Path(PathBuf::from(self)))
```

Comment:
This (and `make_disposable_store`) are the last non-generic storage invocations AFAICT. As `PlatformStorage` could be a non-primitive (like `IpfsStorage` wrapper), some options:

* `Storage` adds a `new_from_path` method that works on all Storage impls (or `new_from_str`)
* `noosphere::platform` exposes something like `make_platform_storage(&str)`, though we want to distinguish `PlatformStorage` (which may be wrapped by a non-primitive store) from the underlying storage primitive, though currently this would involve a lot of duplicate defs if we want to swap around backends during testing, currently apple/aarch64 and non-native-apple multiplied by storage backends